### PR TITLE
Improve Invalid Action return in CommandHandler

### PR DESCRIPTION
**Pull request by yunhanw-google**
#### Problem
https://github.com/project-chip/connectedhomeip/issues/21840

#### Change overview
Improve the error reporting from ProcessGroupCommandDataIB and ProcessCommandDataIB: per spec some of their failures should also be InvalidAction.

#### Testing
Existing test covers

### CODE REVIEW

File: src/app/CommandHandler.cpp [hunk 1]
```diff
@@ -127,14 +127,19 @@ Status CommandHandler::ProcessInvokeRequest(System::PacketBufferHandle && payloa
...
         if (mExchangeCtx->IsGroupExchangeContext())
         {
-            VerifyOrReturnError(ProcessGroupCommandDataIB(commandData) == CHIP_NO_ERROR, Status::Failure);
+            status = ProcessGroupCommandDataIB(commandData);
         }
         else
         {
-            VerifyOrReturnError(ProcessCommandDataIB(commandData) == CHIP_NO_ERROR, Status::Failure);
+            status = ProcessCommandDataIB(commandData);
+            ;
```
Comment (robszewczyk) [inline src/app/CommandHandler.cpp:138]: Unnecessary `;`

File: src/app/CommandHandler.cpp [hunk 2]
```diff
@@ -266,7 +271,11 @@ CHIP_ERROR CommandHandler::ProcessCommandDataIB(CommandDataIB::Parser & aCommand
             ChipLogDetail(DataManagement, "No command " ChipLogFormatMEI " in Cluster " ChipLogFormatMEI " on Endpoint 0x%x",
                           ChipLogValueMEI(concretePath.mCommandId), ChipLogValueMEI(concretePath.mClusterId),
                           concretePath.mEndpointId);
-            return AddStatus(concretePath, commandExists);
+            if (AddStatus(concretePath, commandExists) != CHIP_NO_ERROR)
```
Comment (robszewczyk) [inline src/app/CommandHandler.cpp:274]: style/nit: I'd favor a more compact syntax for this pattern:

````
    return AddStatus(concretePath, commandExists) != CHIP_NO_ERROR ? Status::Failure : Status::Success;
````

Comment (yunhanw-google) [reply src/app/CommandHandler.cpp]: done, thanks

File: src/app/CommandHandler.cpp (hunk 1 above)
Comment (yunhanw-google) [reply src/app/CommandHandler.cpp]: done, thanks

**Review by robszewczyk — APPROVED** [review]

**Review by mrjerryjohns — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

**Pull request merged**